blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs

```diff
@@ -82,5 +82,68 @@ public static async Task CopyBlobAsync(BlobServiceClient blobServiceClient)
             }
         }
         // </Snippet_AbortBlobCopy>
+
+        //-------------------------------------------------
+        // Copy a snapshot over a base blob
+        //-------------------------------------------------
+        // <Snippet_CopySnapshot>
+        public static async Task<BlobClient> CopySnapshotOverBaseBlobAsync(BlobClient blobClient, string snapshotTimestamp)
+        {
+            // Instantiate BlobClient with identical URI and add snapshot timestamp
+            BlobClient blobSnapshotClient = blobClient.WithSnapshot(snapshotTimestamp);
+
+            // Restore the specified snapshot by copying it over the base blob
+            CopyFromUriOperation copyOperation = await blobClient.StartCopyFromUriAsync(blobSnapshotClient.Uri);
+            await copyOperation.WaitForCompletionAsync();
+
+            // Return the client object after the copy operation
+            return blobClient;
+        }
+        // </Snippet_CopySnapshot>
+
+        //-------------------------------------------------
+        // Copy a previous version over a base blob
+        //-------------------------------------------------
+        // <Snippet_CopyVersion>
+        public static async Task<BlobClient> CopyVersionOverBaseBlobAsync(BlobClient blobClient, string versionTimestamp)
+        {
+            // Instantiate BlobClient with identical URI and add version timestamp
+            BlobClient blobVersionClient = blobClient.WithVersion(versionTimestamp);
+
+            // Restore the specified snapshot by copying it over the base blob
+            CopyFromUriOperation copyOperation = await blobClient.StartCopyFromUriAsync(blobVersionClient.Uri);
+            await copyOperation.WaitForCompletionAsync();
+
+            // Return the client object after the copy operation
+            return blobClient;
+        }
+        // </Snippet_CopyVersion>
+
+        //-------------------------------------------------
+        // Rehydrate a blob using a copy operation
+        //-------------------------------------------------
+        // <Snippet_RehydrateUsingCopy>
+        public static async Task RehydrateBlobUsingCopyAsync(BlobServiceClient blobServiceClient)
```

Comment:
I guess I"m not sure why this sample has a different structure than the first. I imagine it could just take `srcArchiveBlob` and `destRehydratedBlob` arguments instead of people wondering why we're creating clients in the sample this time and not others.